### ISSUE TITLE
Cruft cleanup

### ISSUE THREAD

**Issue by joefowler** [usnistgov/mass#100]
**[Original report](https://bitbucket.org/joe_fowler/mass/issues/100) by Joseph Fowler (Bitbucket: [joe_fowler](https://bitbucket.org/joe_fowler), ).**

----------------------------------------

As part of Mass version 0.6, I think we should work to remove **cruft** (see [definition](https://en.wikipedia.org/wiki/Cruft), code that we no longer use). In fact it's mostly code that we, the developers (@oneilg, @jojo978), no longer even expect to work.

The challenge is that some function/method that *I think* to be unused is actually used *all the time* by someone else. 

The following is a far-from-complete list, but some items that seem to me to be no longer useful include:

1. MicrocalDataSet.fit_spectral_line: *actually, this is used; leave it in*
1. ~~Pickling of calibration data: not generally used~~ DONE
1. ~~ MicrocalDataSet.time_drift_correct_polynomial: never made to work~~ DONE
1. ~~Per-channel gains: a single scaling factor just isn't enough!~~ DONE
1. ~~mass.gui.compute_gains: a never-finished GUI to compute gains (which are not useful)~~ DONE
1. ~~TESGroup.find_features_with_mouse~~ DONE
1. ~~TESGroup.join: Galen thinks it doesn't work~~ DONE
1. ~~mass.calibration.spectra: a module we don't use~~ DONE

Am I right? Or are some of these used by other people?

**Comment by joefowler** [comment]
**Original comment by Galen O'Neil (Bitbucket: [oneilg](https://bitbucket.org/oneilg), GitHub: [oneilg](https://github.com/oneilg)).**

----------------------------------------

I was going to write that I still use pickling of Calibrations, but I don't see that in the code anymore. It looks like they're written to an hdf5 group. But I couldn't find any code that can load them from an HDF5 group? but that they aren't loaded back in? I'm asking mostly because this is probably going to come up at the SSRL today.

FWIW the only occurrences of the word "pickle" in either `channel.py` or `channel_group.py` appear to be imports.

**Comment by joefowler** [comment]
**Original comment by Galen O'Neil (Bitbucket: [oneilg](https://bitbucket.org/oneilg), GitHub: [oneilg](https://github.com/oneilg)).**

----------------------------------------

For number 7 a more correct statement is "Galen thinks this is the first time he's ever heard of TESGroup.join". Maybe your statement was correct in the past.

**Comment by joefowler** [comment]
**Original comment by Joseph Fowler (Bitbucket: [joe_fowler](https://bitbucket.org/joe_fowler), ).**

----------------------------------------

Yep, and if you search for "pkl", you don't get anything but a utility function to generate filenames.

If you find that calibrations are stored to HDF5 but not loadable, then that's definitely interesting (and worth its own separate issue).

**Comment by joefowler** [comment]
**Original comment by Joseph Fowler (Bitbucket: [joe_fowler](https://bitbucket.org/joe_fowler), ).**

----------------------------------------

Well, this is probably my fault, but inside the `TESGroup.join()` method is

```
        print('join probably doesnt work since galen messed with it moving things inside datasets')
```

**Comment by joefowler** [comment]
**Original comment by Galen O'Neil (Bitbucket: [oneilg](https://bitbucket.org/oneilg), GitHub: [oneilg](https://github.com/oneilg)).**

----------------------------------------

Let me take a crack at 2 then, it will help me refresh myself for today and clean stuff up.

**Comment by joefowler** [comment]
**Original comment by Joseph Fowler (Bitbucket: [joe_fowler](https://bitbucket.org/joe_fowler), ).**

----------------------------------------

Okay, I will work on 4-8. Mostly I expect to hide the larger ones in `src/mass/nonstandard` so we can work on any of them later, in the unlikely event that we want to.

Do you think this deserves its own branch? Not sure how I want to handle that.

**Comment by joefowler** [comment]
**Original comment by Galen O'Neil (Bitbucket: [oneilg](https://bitbucket.org/oneilg), GitHub: [oneilg](https://github.com/oneilg)).**

----------------------------------------

Probably not.

**Comment by joefowler** [comment]
**Original comment by Galen O'Neil (Bitbucket: [oneilg](https://bitbucket.org/oneilg), GitHub: [oneilg](https://github.com/oneilg)).**

----------------------------------------

Remove pickle related cruft, see #100

**Comment by joefowler** [comment]
**Original comment by Joseph Fowler (Bitbucket: [joe_fowler](https://bitbucket.org/joe_fowler), ).**

----------------------------------------

No one has touched spectra.py in years. Deprecate it!

(No substantive changes since 2012. I think it’s unused.
See #100.)

**Comment by joefowler** [comment]
**Original comment by Joseph Fowler (Bitbucket: [joe_fowler](https://bitbucket.org/joe_fowler), ).**

----------------------------------------

@jojo978 : do you use any of these objects and/or want to preserve them?

**Comment by joefowler** [comment]
**Original comment by Joseph Fowler (Bitbucket: [joe_fowler](https://bitbucket.org/joe_fowler), ).**

----------------------------------------

Remove MicrocalDataSet.time_drift_correct_polynomial()

It was not working. Find in src/nonstandard/drift_correction.py.
Fixes #100 finally!
